src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -856,12 +832,12 @@ private void moveUnitsToDefendTerritories(final Map<Territory, ProTerritory> mov
           final List<Unit> defendingUnits = moveMap.get(t).getAllDefenders();
           if (moveMap.get(t).getBattleResult() == null) {
             moveMap.get(t).setBattleResult(
-                battleUtils.estimateDefendBattleResults(player, t, moveMap.get(t).getMaxEnemyUnits(), defendingUnits,
-                    moveMap.get(t).getMaxEnemyBombardUnits()));
+                ProBattleUtils.estimateDefendBattleResults(player, t, moveMap.get(t).getMaxEnemyUnits(),
```

Comment:
Duplicated code, lines 835 and 790. Looks like making `moveMap` be a first class data would help fix this. Then the `moveMap` would interact with ProBattleUtils, and the `moveMap` could help offload the functionality found in this long method